ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/MergeTransitionConfigCheckTest.java

```diff
@@ -0,0 +1,145 @@
+import tech.pegasys.teku.spec.config.SpecConfigLoader;
+import tech.pegasys.teku.spec.executionengine.ExecutionEngineChannel;
+import tech.pegasys.teku.spec.executionengine.TransitionConfiguration;
+import tech.pegasys.teku.spec.util.DataStructureUtil;
+
+class MergeTransitionConfigCheckTest {
+
+  private static final UInt64 BELLATRIX_FORK_EPOCH = UInt64.ONE;
+  private static final Bytes32 TERMINAL_BLOCK_HASH = Bytes32.random();
+  private static final UInt64 TERMINAL_BLOCK_EPOCH = UInt64.valueOf(2);
+
+  private final ExecutionEngineChannel executionEngine = mock(ExecutionEngineChannel.class);
+  private final StubTimeProvider timeProvider = StubTimeProvider.withTimeInSeconds(10_000);
+  private final StubAsyncRunner asyncRunner = new StubAsyncRunner(timeProvider);
+  private final EventLogger eventLogger = Mockito.mock(EventLogger.class);
+
+  private final Spec spec =
+      TestSpecFactory.createBellatrix(
+          SpecConfigLoader.loadConfig(
+              "minimal",
+              phase0Builder ->
+                  phase0Builder
+                      .altairBuilder(altairBuilder -> altairBuilder.altairForkEpoch(UInt64.ZERO))
+                      .bellatrixBuilder(
+                          bellatrixBuilder ->
+                              bellatrixBuilder
+                                  .bellatrixForkEpoch(BELLATRIX_FORK_EPOCH)
+                                  .terminalBlockHash(TERMINAL_BLOCK_HASH)
+                                  .terminalBlockHashActivationEpoch(TERMINAL_BLOCK_EPOCH))));
+
+  private final DataStructureUtil dataStructureUtil = new DataStructureUtil(spec);
+  private TransitionConfiguration localTransitionConfiguration;
+
+  private final UInt256 wrongRemoteTTD = dataStructureUtil.randomUInt256();
+  private final Bytes32 wrongRemoteTBH = dataStructureUtil.randomBytes32();
+
+  @BeforeEach
+  void setUp() {
+    localTransitionConfiguration =
+        new TransitionConfiguration(
+            spec.getGenesisSpecConfig()
+                .toVersionBellatrix()
+                .orElseThrow()
+                .getTerminalTotalDifficulty(),
+            spec.getGenesisSpecConfig().toVersionBellatrix().orElseThrow().getTerminalBlockHash(),
+            UInt64.ZERO);
+
+    final MergeTransitionConfigCheck mergeTransitionConfigCheck =
+        new MergeTransitionConfigCheck(eventLogger, spec, executionEngine, asyncRunner);
+
+    when(executionEngine.exchangeTransitionConfiguration(localTransitionConfiguration))
+        .thenReturn(
+            SafeFuture.completedFuture(
+                new TransitionConfiguration(
+                    localTransitionConfiguration.getTerminalTotalDifficulty(),
+                    localTransitionConfiguration.getTerminalBlockHash(),
+                    dataStructureUtil.randomUInt64())));
+
+    assertThat(mergeTransitionConfigCheck.start()).isCompleted();
+  }
+
+  @Test
+  void shouldReportWrongTotalTerminalDifficulty() {
+    // wrong terminal total difficulty
```

Comment:
now that we've got the test names all pretty can eliminate the comments, but thats admittedly a super nit...

Reply:
Yeah makes sense.